src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -57,14 +62,21 @@ public NestedFieldProjection(SearchRequestBuilder request) {
     public void project(List<Field> fields) {
         if (isAnyNestedField(fields)) {
             initBoolQueryFilterIfNull();
-
+            // Extract nested query from WHERE clause
             List<NestedQueryBuilder> nestedQueries = extractNestedQueries(query());
             groupFieldNamesByPath(fields).forEach(
-                (path, fieldNames) -> buildInnerHit(fieldNames, findNestedQueryWithSamePath(nestedQueries, path))
+                (path, fieldNames) -> buildInnerHit(fieldNames, findNestedQueryWithSamePath(nestedQueries, path, fieldNames))
```

Comment:
Could you add comment, explaining what's going on here?

Reply:
will add documentation to class level as dai-chen pointed out. https://github.com/opendistro-for-elasticsearch/sql/pull/153#discussion_r314913904

Reply:
will remove the fieldNames, will clean the inner logic and then no need to pass this variable.